### PR TITLE
Refactor to contain resident and help request

### PR DESCRIPTION
**Pull request by kosiakkatrina**
# What

Added Resident entity, caseNotes entity and new HelpRequest entity

Refactor help request gateway into:

- HelpRequestGateway
- ResidentGateway
- CaseNotesGateway

Refactor **createHelpRequestUseCase** into **CreateResidentAndHelpRequestUseCase** which calls **createResidentUseCase** (calls three gateway methods findResident, updateResident-if resident exists and data has changed, and createResident), **createHelpRequestUseCase**(calls a single createHelpRequest gateway method) and **CreateCaseNotesUseCase**

Refactor **UpdateHelpRequestUsecase** into **UpdateResidentAndHelpRequestUseCase** in a similar way as create use case

Refactor **PatchHelpRequestUsecase** into **PatchResidentAndHelpRequestUseCase** in a similar way as create use case

Rename GetHelpRequestUseCase to GetResidentAndHelpRequestUseCase, update getHelpRequest gateway method to return resident information as well by using navigational properties

Rename GetHelpRequestsUseCase to GetResidentsAndHelpRequestsUseCase, update searchHelpRequest gateway method to return resident information as well by using navigational properties

Update getCallbacks gateway method to return resident information as well by using navigational properties

Add The following domain objects:
image
**HelpRequestWithResident** domain object holds both info about resident and help request
**Resident** domain object only holds the resident info
**HelpRequest** domain object only holds the help request info

**CreateResidentAndHelpRequest** command object holds all of the info that comes from the current create endpoint
**CreateResident** command object only holds the resident info
**CreateHelpRequest** command object only holds the help request info

Similar structure for Patch and update commands

Also add/remove relevant factory mappings.

# Why
To enable the API talk to the new database schema

# Screenshots

# Link to ticket
https://trello.com/c/sOF9V4DU/188-hth-3-migration-and-logic-gateway-changes

# Notes
Need to add deduplication for help requests, probably by CTAS id, can do in a separate PR

### CODE REVIEW

**Review by antonyoneill — CHANGES_REQUESTED** [review]
Just I think two things that I've highlighted that I think need to be changed before we can get this merged, otherwise on the whole this is very good @kosiakkatrina @maysakanoni !!! 

Some great work.

Tl;dr:
Not to act on in this PR:
* Couple of minor changes in terms of parameter naming convention, and ordering of parameters
* Some `if` blocks could be refactored to read a little easier
* Couple of missing implementations for some usecases, but I think they're coming soon
* I thought we were storing the resident DOB as a date, not separate fields, but I'm not too concerned about this at the moment as it will have wider implications
* There's some confusion around PATCH and PUT - let's discuss
* Next time, let's sit together and figure out how to get this broken down into smaller PRs - there are some natural lines that I can see that could be good splits

To act on in this PR (these are marked as [Should change]:
* Responsibility of adding case notes for a new HelpRequest should be moved into the `CreateHelpRequestUseCase`
* Resident duplication logic should be case insensitive for names

All in all, you've both done a great job, let's do the final push to get this over the line

File: cv19ResSupportV3/V3/Gateways/ResidentGateway.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                    }
+                }
+
+                if (command.DobDay != null || command.DobMonth != null || command.DobYear != null)
+                {
+                    var response = _helpRequestsContext.ResidentEntities.FirstOrDefault(x =>
+                        x.DobDay == command.DobDay &&
+                        x.DobMonth == command.DobMonth &&
+                        x.DobYear == command.DobYear &&
+                        x.FirstName == command.FirstName &&
```
Comment (antonyoneill) [inline cv19ResSupportV3/V3/Gateways/ResidentGateway.cs:190]: Case sensitive

Comment (antonyoneill) [reply cv19ResSupportV3/V3/Gateways/ResidentGateway.cs:190]: + trim pls

**Review by antonyoneill — APPROVED** [review]
